### PR TITLE
Added support for passing node list line as argument to scripts.

### PR DESCRIPTION
**Pull request by jeffmontagna**
- Added support to rolling-restart.sh to pass a line from the node list to the startup and shutdown scripts.
- Modified example scripts to include parsing of a comma delimited argument.
- Updated README to reflect these changes.

### CODE REVIEW

File: rolling-restart.sh [hunk 1]
```diff
@@ -115,7 +113,7 @@ for NODE in ${NODES[@]}; do
 
     echo ">>>>>> Running shutdown script for ${NODE}"
 
-    eval $SHUTDOWN_SCRIPT
+    eval $SHUTDOWN_SCRIPT $NODE_CFG
```
Comment (erik-stephens) [inline rolling-restart.sh:119]: I'm curious why the `eval` is needed.

File: README.md [hunk 2]
```diff
@@ -6,22 +6,22 @@ This is a simple script to do a controlled rolling restart of a multinode elasti
...
+* -s - SCRIPT - Script to run on each node to process the update.
 
 ```
 # ./rolling-restart.sh -m master1.example.com:9200 -n ./node-list.txt -s ./node-update.sh
 ```
 
 ### NODE_FILE
 
-The node file is a list of node names (including HTTP port) one per line. These should only be nodes that contain data and the node list should not contain the host you're using as the master node.
+The node file is a list of node names (including HTTP port) one per line. These should only be nodes that contain data and the node list should not contain the host you're using as the master node.  Each line in the node file can contain a comma delimted list of paramaters to pass in to the startup and shutdown scripts.
```
Comment (godber) [inline README.md]: I'd do a little rework on this section of the readme and keep the example of a minimal node list

```
node1.example.com:9200
node2.example.com:9200
node3.example.com:9200
```

and then explain how the expanded node list can be used.

Erik simplified this for cases where each node needs the same variables by just exporting them in the calling context and then using them within the update script ... or something along those lines.  I think the expanded node_cfg could still be useful when nodes don't share values.

File: README.md [hunk 3]
```diff
@@ -30,18 +30,36 @@ The script should contain the actions to perform on the stopped node. This assum
 
 When the script runs the Elasticsearch instance on the node will already be stopped. It is the script's responsibility to restart it once updates are complete.
 
-Two variables are made available to the script.
-
-* $MASTER is the master node as provided on the command line.
-* $NODE is the name of the node being restarted. This name will include the port.
+The comma seperated list for a single node from the node_file is passed into the scirpt as an argument.
```
Comment (godber) [inline README.md]: scirpt -> script

File: example-node-list.txt [hunk 4]
```diff
@@ -1,3 +1,3 @@
-node1.example.com:9200
-node2.example.com:9200
-node3.example.com:9200
\ No newline at end of file
+node1.example.com:9200,cluster1,elasticsearch,elasticsearch_cluster1_dev
+node2.example.com:9200,cluster1,elasticsearch,elasticsearch_cluster1_dev
+node3.example.com:9200,cluster1,elasticsearch,elasticsearch_cluster1_dev
```
Comment (godber) [inline example-node-list.txt]: Maybe leave `example-node-list.txt` untouched and add a `example-node-list-expanded.txt`.

**Comment by peterdemartini** [comment]
Is this still valid?

**Comment by godber** [comment]
Probably not, @jeffmontagna, lets close this if it's not really relevant internally.

**Pull request merged**